src/renderers/shaders/ShaderChunk/global_fragment.glsl

```diff
@@ -0,0 +1 @@
+// template for injecting varying, uniform and functions above main below other includes
```

Comment:
I read the other one again. I can see it as actually being the ultimate abstraction, "here's the result of the template for you to alter right before we compile for you", but the argument i'd make is - another system is being introduced, when it seems like the existing uniforms one would suffice. But, the hooks seem to take care of this with much less code, so it seems like a win.